ide/api.lsp/src/org/netbeans/api/lsp/StructureElement.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.api.lsp;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * StructureElement is a tree item that shows the structure of the source code.
+ * 
+ * @author Petr Pisl
+ * @since 1.8
+ */
+public interface StructureElement {
```

Comment:
How do you want to evolve this type? You may need to add new getters in the future.

I't be better to follow the pattern of [Completion](https://bits.netbeans.org/dev/javadoc/org-netbeans-api-lsp/org/netbeans/api/lsp/Completion.html) in API and 
[CompletionCollector](https://bits.netbeans.org/dev/javadoc/org-netbeans-api-lsp/org/netbeans/spi/lsp/CompletionCollector.html) in SPI.

Turn `StructureElement` into `final class`.

Reply:
I was trying to create the sub nodes lazily. The tree doesn't appear all at once, only the top two levels. But in the end, the whole tree is sent in response, so from that perspective it doesn't matter. But for the future ....

Reply:
Done